core/backend/src/rpc-impl/IModelReadRpcImpl.ts

```diff
@@ -194,6 +194,37 @@ export class IModelReadRpcImpl extends RpcInterface implements IModelReadRpcInte
     return iModelDb.getMassProperties(props);
   }
 
+  public async getMassPropertiesPerCandidate(tokenProps: IModelRpcProps, props: MassPropertiesPerCandidateRequestProps): Promise<MassPropertiesPerCandidateResponseProps[]> {
+    const iModelDb = await RpcBriefcaseUtility.findOpenIModel(RpcTrace.expectCurrentActivity.accessToken, tokenProps);
+    const result: MassPropertiesPerCandidateResponseProps[] = [];
+
+    for (const candidate of CompressedId64Set.iterable(props.candidates)) {
+      const massPropResults: MassPropertiesResponseProps[] = [];
+
+      for (const op of props.operations) {
+        const massProperties = await iModelDb.getMassProperties({ operation: op, candidates: [candidate] });
+        massPropResults.push(massProperties);
+      }
+
+      let singleCandidateResult: MassPropertiesPerCandidateResponseProps = { status: BentleyStatus.ERROR, candidate };
+
+      if (massPropResults.some((r) => r.status !== BentleyStatus.ERROR)) {
+        singleCandidateResult.status = BentleyStatus.SUCCESS;
+        for (const r of massPropResults.filter((mpr) => mpr.status !== BentleyStatus.ERROR)) {
+          singleCandidateResult = { ...singleCandidateResult, ...r };
+        }
+      }
+
+      result.push(singleCandidateResult);
+
+      if (result.length === 1000) {
```

Comment:
So there's a hard-coded limit of 1000 candidates per request? Document it. Why 1000?

Reply:
The consuming application works in batches of 1000 candidates max, so I thought I'd have the same limit here. I was afraid that allowing unlimited amount of candidates could lead to denial of service. `getMassProperties` is relatively expensive and could take a while if millions of candidates are passed in.

Do you think I should remove the limit?

Reply:
I'm asking what motivated the specific limit of 1000 (some measurement that makes this a good choice over some other number?) and requesting that any such limit be clearly documented and enforced (e.g., you should probably throw an exception if the limit is exceeded instead of silently truncating the results).
We should not design our library to cater to one specific consuming application.

Reply:
There were no measurements. The only motivation was the limit on client application, therefore removed the limit. I found an example of other operation (`getGeometrySummary`) that accept an unlimited amount of elementIds.